datasets/BEIS-Renewables-obligation-certificates-and-generation-monthly/main.py

```diff
@@ -0,0 +1,124 @@
+    trace.Generation_Type("Defined from cell B6 below which is not blank")
+
+    roc_per_mwh = cell.shift(2, 5).fill(DOWN).is_not_blank().is_not_whitespace()
+    trace.Roc_Per_Mwh("Defined from cell C6 below which is not blank")
+    
+    if tab.name == "Quarter":
+        qtr = cell.shift(2, 5).fill(RIGHT).is_not_blank().is_not_whitespace()
+        trace.Qtr("Defined from cell D6 and right which is not blank")
+        period = cell.shift(2, 4).fill(RIGHT).is_not_blank().is_not_whitespace()
+        trace.Period("Defined from cell D5 and right which is not blank")
+    elif tab.name == "Month":
+        month = cell.shift(2, 5).fill(RIGHT).is_not_blank().is_not_whitespace()
+        trace.Month("Defined from cell D6 and right which is not blank")
+        period = cell.shift(2, 4).fill(RIGHT).is_not_blank().is_not_whitespace()
+        trace.Period("Defined from cell D5 and right which is not blank")
+    else:
+        period = cell.shift(2, 5).fill(RIGHT).is_not_blank().is_not_whitespace()
+        trace.Period("Defined from cell D6 and right which is not blank")
+
+    element = tab.one_of(["Certificates", "Equivalent Generation"])
+    trace.Element("Defined cell A7, A132, A257 and A275")
+
+    techno_group = cell.shift(0, 5).fill(DOWN).is_not_blank().is_not_whitespace()-element
+    
+    technology_group = techno_group - tab.excel_ref("A").filter(contains_string("Summary Technology Group"))
+    trace.Technology_Group("Defined from cell A8 which is not blank")
+    
+    observations = cell.shift(3, 6).expand(RIGHT).expand(DOWN).is_not_whitespace()
+# ROCs and MWh needs to be removed from the observation which is getting caught in DATAMARKER.
+# true_observations = observations.filter(lambda x: type(x.value) != "ROCs" not in x.value)
+
+    if tab.name == "Quarter": 
+    
+        dimensions = [
+            HDim(technology_group, "Technology Group", CLOSEST, ABOVE),
+            HDim(generation_type, "Generation Type", CLOSEST, ABOVE),
+            HDim(roc_per_mwh, "Roc Per Mwh", CLOSEST, ABOVE),
+            HDim(period, "Period", CLOSEST, LEFT),
+            HDim(element, "Element", CLOSEST, ABOVE),
+            HDim(qtr, "Qtr", CLOSEST, LEFT)
+        ]
+        
+    elif tab.name == "Month":
+        
+        dimensions = [
+            HDim(technology_group, "Technology Group", CLOSEST, ABOVE),
+            HDim(generation_type, "Generation Type", CLOSEST, ABOVE),
+            HDim(roc_per_mwh, "Roc Per Mwh", CLOSEST, ABOVE),
+            HDim(period, "Period", CLOSEST, LEFT),
+            HDim(element, "Element", CLOSEST, ABOVE),
+            HDim(month, "Month", CLOSEST, LEFT)
+        ]
+    else:
+            
+        dimensions = [
+            HDim(technology_group, "Technology Group", CLOSEST, ABOVE),
+            HDim(generation_type, "Generation Type", CLOSEST, ABOVE),
+            HDim(roc_per_mwh, "Roc Per Mwh", CLOSEST, ABOVE),
```

Comment:
```suggestion
            HDim(roc_per_mwh, "Roc Per Mwh", DIRECTLY, LEFT, cellvalueoverride = {'' : 'all'}),
```

Reply:
DIRECTLY LEFT instead of CLOSEST ABOVE resolves captured values that doesn't belong to roc_per_mwh column and blank cell values in the column is override by 'all'